torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -59,16 +59,24 @@ static std::vector<std::string> TORCH_NCCL_DESYNC_DEBUG = {
     "TORCH_NCCL_DESYNC_DEBUG",
     "NCCL_DESYNC_DEBUG"};
 
+// Environment variable to enable whether to record and check the start
```

Comment:
can remove the "Environment variable" prefix from all of these

"of one collective" (?)  not "all collectives"? 

How about
"Enable recording start-events for all ProcessGroupNCCL collectives, and compute accurate collective timing per-collective.  (Note: end-events are recorded by default).  Can increase chances of a watchdog hang due to querying cudaEventElapsedTime() API"

Reply:
Maybe I should say each collective but using "all" also works.